ccl/changefeedccl: TestAlterChangefeedAddTargetsDuringBackfill failed

ccl/changefeedccl.TestAlterChangefeedAddTargetsDuringBackfill [failed](https://tanzanite.cluster.engflow.com/invocations/default/63d14e26-25b4-4ef6-b012-c175658024c2?testReportRun=1&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on release-24.1 @ [557f4e6fe89d683d49082d88ba5adf6660e8f16f](https://github.com/cockroachdb/cockroach/commits/557f4e6fe89d683d49082d88ba5adf6660e8f16f):


```
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616643821992, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616652518380, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616660308308, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616667869867, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616675161625, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616682339224, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616688874142, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616696340460, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616858657503, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616866298391, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616956386200, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703616978766785, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703617009599988, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/106/1, EndKey:/Table/106/2}, Timestamp:hlc.Timestamp{WallTime:1711703619446658198, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703619446658198, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/106/1, EndKey:/Table/106/2}, Timestamp:hlc.Timestamp{WallTime:1711703622284177045, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703622284177045, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/106/1, EndKey:/Table/106/2}, Timestamp:hlc.Timestamp{WallTime:1711703625309051470, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703625309051470, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/106/1, EndKey:/Table/106/2}, Timestamp:hlc.Timestamp{WallTime:1711703628381213233, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703628381213233, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/106/1, EndKey:/Table/106/2}, Timestamp:hlc.Timestamp{WallTime:1711703630819644193, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/106/1, EndKey:/Table/106/2}, Timestamp:hlc.Timestamp{WallTime:1711703630821468972, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631186632548, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631198139585, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631208489803, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631218681011, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631228269608, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631238571116, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631248986444, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631261445191, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631274604698, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631295809143, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631305271971, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631320237597, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631335065374, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631348736461, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631358843478, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631369913126, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631379621504, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631394667660, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631407366067, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631418245224, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631430667352, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631440984269, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631450103387, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631460138875, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/107/1, EndKey:/Table/107/2}, Timestamp:hlc.Timestamp{WallTime:1711703631472670152, Logical:0}, BoundaryType:0}
    alter_changefeed_test.go:1435: resolved span: &jobspb.ResolvedSpan{Span:roachpb.Span{Key:/Table/106/1, EndKey:/Table/106/2}, Timestamp:hlc.Timestamp{WallTime:1711703631422302784, Logical:0}, BoundaryType:0}
    --- FAIL: TestAlterChangefeedAddTargetsDuringBackfill/enterprise (167.18s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAlterChangefeedAddTargetsDuringBackfill.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37208

--- a/pkg/ccl/changefeedccl/helpers_test.go
+++ b/pkg/ccl/changefeedccl/helpers_test.go
@@ -671,9 +671,9 @@ var jobRecordPollFrequency = 3 * time.Second
 
 var jobRecordRetryOpts = retry.Options{
 	InitialBackoff: jobRecordPollFrequency,
-	Multiplier:     1,
-	MaxBackoff:     5 * time.Second,
-	MaxRetries:     30,
+	Multiplier:     2,
+	MaxBackoff:     1 * time.Minute,
+	MaxRetries:     10,
 }
 
 // waitForCheckpoint waits for the specified job to have a non-empty checkpoint

